client/src/components/DataFiles/DataFilesToolbar/DataFilesToolbar.jsx

```diff
@@ -151,10 +151,17 @@ const DataFilesToolbar = ({ scheme, api }) => {
   };
   const download = () => {
     if (canDownload) {
-      dispatch({
-        type: 'DATA_FILES_DOWNLOAD',
-        payload: { file: selectedFiles[0] },
-      });
+      // Checks to see if the file is less than 2 GB; executes the dispatch if true and displays the Globus alert if false
+      if (selectedFiles[0].length < 2000000000) {
```

Comment:
The number 2000000000 is not easily readable (counting zeros is error prone). A suggestion is to use a constant like this.

```
const MAX_FILE_SIZE = 2 * 1024 * 1024 * 1024; // 2GB in bytes
```

Reply:
Also, does this only check the first file in the list ? or the sum of all the files that are getting zipped ?

Reply:
I'll make the change you listed above.

When I tested this, I zipped up a folder and found its size to be 5.6 GB, so I'm pretty sure that the file size is size of the zipped folder.

Reply:
This does not resolve the actual issue. The task was opened when we ran into an issue with a user trying to download a folder with a million files (we do not want to repro this and bring down the system now). I updated the root cause in the task. Below is root cause. The fix above does not handle the root cause. It disallows download of a zip file which is greater than max file size. But, the damage happens even before the zip is created.


When more than 1 file is involved in download, the following happens:

1. Portal triggers a compress job. 
2. The [compress job ](https://github.com/TACC/WMA-Tapis-Templates/blob/main/applications/compress/app.json)used right now, runs in frontera and copies all the files from source to frontera scratch.  
3. It then zip those files and provides them for download to the user. 
4. During step 2) above - if there are 1000s of files or files compressed is large, it could cause DoS (denial of service) for other file operations in TAPIS and SSH problems with copying one file at a time from corral to frontera.

Most of the UI side of the logic is in DataFilesDownloadMessageModal.